### PR TITLE
[stable/concourse] Add rbac to concourse worker

### PR DESCRIPTION
**Pull request by nexeck**
This will add rbac support to concourse

### CODE REVIEW

**Comment by william-tran** [comment]
@nexeck I have another PR #2911 that would also benefit from RBAC, but concourse-web needs a serviceAccount separate from the workers, as it (and not the workers), will need access to secrets in namespaces used solely for concourse. Would it be possible to include concourse-web as well in your PR?

**Comment by nexeck** [comment]
Sure, is it enough to add this roles to the web serviceaccount?

```
rules:
- apiGroups: [""]
  resources: ["secrets"]
  verbs: ["get", "watch", "list"]
```

**Comment by william-tran** [comment]
Thanks @nexeck, just `get` is required. Here's the lookup scheme for background: https://github.com/autonomic-ai/charts/blob/8d038734d8a7f4421356ae5f324ca3f5b6d382dd/stable/concourse/README.md#kubernetes-secrets

**Comment by nexeck** [comment]
So the web worker needs this as a clusterrole to access other namespaces?

**Comment by william-tran** [comment]
Yes, the set of namespaces could default to just `[credentialManager.kubernetes.namespacePrefix]main` as main is the team that you start with and you can't control the teams that get created from the chart.

**Comment by nexeck** [comment]
I would suggest something like this:
```
rbac:
  create: false

  ## Ignored if rbac.create is true
  ##
  serviceAccountName: default

  teams:
  - main
```

Maybe the teams would better fit something else in the configuration?

**Comment by william-tran** [comment]
So the intent is that whenever a new team is created, that a new entry is added under rbac.teams (or a more appropriate key)?

**Comment by nexeck** [comment]
Yes, that is the intention, otherwise the web SA needs to be able to read all secrets in all namespaces.
Do you have other ideas, how we could solve this?

**Comment by william-tran** [comment]
No, I think its a totally reasonable approach to enumerate the teams. I
need to give a bit more thought to what key that list should fall under.

**Comment by william-tran** [comment]
@nexeck I've added a key in this commit https://github.com/kubernetes/charts/pull/2911/commits/4b36ab4fbd4e28a6222a43ec99f01fec924c2a31

#2911 now depends on RBAC, so let me know when you're done with this PR and I'll add those commits, retaining authorship of course, to #2911. Or you can make a PR to my fork.

**Comment by nexeck** [comment]
@william-tran Can you take a look at 2a53d44 ?
Haven't tested this against your branch.

**Comment by william-tran** [comment]
@nexeck I found a few errors, corrected in this commit https://github.com/autonomic-ai/charts/commit/bd93bd6891e124a523e0124e597634193325c6be I'll let you know once I get this tested with `rbac.create: true`

**Comment by william-tran** [comment]
@nexeck Since there are issues that need to be fixed in concourse to support #2911, lets consider this PR first. What are the motivations for adding RBAC to the workers only?

**Comment by nexeck** [comment]
@william-tran the worker needs it to be able to run with privileged: true. Without this the worker are not able to run any jobs.
I can add it for the web, too. But at the moment i do not know if the web needs any role. In our current setup it is working fine without giving the web any special rights.

But for completion, i would add it for the web, too.
Do you know, if i need to give it any role (Does the default SA have any rights/roles?)?
In our current setup it is running with the default SA.

**Comment by william-tran** [comment]
@nexeck I would just provide the workers with the RBAC to have it run with `privileged: true` ad leave the web one alone, once I get the k8s secrets stuff working, tested and stable I'll add RBAC for concourse-web.

**Comment by nexeck** [comment]
@william-tran  i have removed the web rbac commit

/assign @viglesiasce

File: stable/concourse/values.yaml [hunk 1]
```diff
@@ -572,3 +572,11 @@ credentialManager:
     ## the value will be written to /concourse-vault/client.key
     ## make sure to also set authBackend to `cert`
     # clientKey: |-
+
+## For RBAC support:
+rbac:
+  create: false
```
Comment (viglesiasce) [inline stable/concourse/values.yaml]: Most other charts have this as `enabled`. Can you make the change?

**Review by viglesiasce — CHANGES_REQUESTED** [review]
Only one nit on the value name

**Comment by nexeck** [comment]
@viglesiasce i have renamed the key from create to enabled

**Review by viglesiasce — APPROVED** [review]

**Comment by unguiculus** [comment]
@viglesiasce Actually, our best practices suggest to use `rbac.create`. There's been quite a bunch of PRs now that follow this pattern. See https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md. 

@nexeck Care to open another PR to fix this?

**Comment by nexeck** [comment]
@unguiculus Sure, if this PR is merged, i will open a new one, with rbac.create.

**Comment by prydonius** [comment]
@mattfarina any idea why this one isn't getting auto-merged? Does it need `/ok-to-test`?

**Comment by unguiculus** [comment]
@nexeck I thought this had been merged already. So you can fix it directly in this PR. I guess I got confused by the comment of the bot.

**Comment by nexeck** [comment]
@unguiculus i have removed the last commit which switched from rbac.create to rbac.enabled

**Comment by nexeck** [comment]
rbac.create is per default now true, like suggested in: https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

**Comment by nexeck** [comment]
Doesn't the check runs with k8s 1.8?

**Comment by unguiculus** [comment]
We support compatibility with the current and previous minor version, which would be 1.7+. The chart has to work on 1.7.

**Comment by nexeck** [comment]
@unguiculus i have added a value for the rbac version, the default is now v1beta1

**Comment by nexeck** [comment]
@unguiculus is anything else missing here?

**Pull request merged**